cwl/src/main/scala/wdl4s/cwl/Decoder.scala

```diff
@@ -0,0 +1,53 @@
+package wdl4s.cwl
+
+import cats.data.{EitherT, NonEmptyList, ValidatedNel}
+import ammonite.ops._
+import ammonite.ops.ImplicitWd._
+import cats.effect.IO
+import cats.syntax.either._
+import better.files.{File => BFile}
+
+import scala.util.Try
+
+object CwlDecoder {
+
+  type Parse[A] = EitherT[IO, NonEmptyList[String], A]
+  type ParseValidated[A] = IO[ValidatedNel[String, A]]
+
+  def salad(path: BFile): Parse[String] = {
```

Comment:
It's probably not clear to most readers that `salad` is being used a verb in the imperative mood; a different name might help.  🙂